quolance-api/src/main/java/com/quolance/quolance_api/controllers/BlogPostController.java

```diff
@@ -0,0 +1,65 @@
+package com.quolance.quolance_api.controllers;
+
+import com.quolance.quolance_api.dtos.blog.BlogPostRequestDto;
+import com.quolance.quolance_api.dtos.blog.BlogPostResponseDto;
+import com.quolance.quolance_api.services.entity_services.BlogPostService;
+import io.swagger.v3.oas.annotations.Operation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import jakarta.validation.Valid;
+import java.util.List;
+
+
+
+@RestController
+@RequestMapping("/api/blog-posts")
+@RequiredArgsConstructor
+public class BlogPostController {
+
+    private final BlogPostService blogPostService;
+
+    @PostMapping
+    @Operation(summary = "Create a blog post")
+    public ResponseEntity<BlogPostResponseDto> createBlogPost(@Valid @RequestBody BlogPostRequestDto request) {
+        BlogPostResponseDto response = blogPostService.create(request);
```

Comment:
need to pass in authenticated user to the create function.
Something like : `User poster = SecurityUtil.getAuthenticatedUser();`